semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -423,7 +427,22 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
       return Optional.empty();
     }
 
-    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    return getHttpSchemeFromRawAttributes(event.getAttributes().getAttributeMap());
+  }
+
+  private static Optional<String> getHttpSchemeFromRawAttributes(
+      Map<String, AttributeValue> attributeValueMap) {
+    // dealing with the Forwarded header separately as it may have
+    // more info than just the protocol
+    if (attributeValueMap.get(HTTP_REQUEST_FORWARDED.getValue()) != null
```

Comment:
We can avoid multiple get calls.

```
AttributeValue attributeValue = attributeValueMap.get(HTTP_REQUEST_FORWARDED_ATTR);
    if (attributeValue != null && !StringUtils.isEmpty(attributeValue.getValue())) {
      String schemeValue = attributeValue.getValue();
      Optional<String> optionalExtractedProtoValue = getProtocolFromForwarded(schemeValue);
      if (optionalExtractedProtoValue.isPresent()) {
        return optionalExtractedProtoValue;
      }
    }
```
And we can define static var at the class level avoiding multiple conversation everytime its get called.
```
private static final String HTTP_REQUEST_FORWARDED_ATTR = HTTP_REQUEST_FORWARDED.getValue();
```

Reply:
Refactored the code. Now `HTTP_REQUEST_FORWARDED.getValue()` is only used at one place so not defining any static var for this.